lib/std/treap.zig

```diff
@@ -447,3 +455,194 @@ test "insert, find, replace, remove" {
         try testing.expectEqual(entry.node, treap.getEntryFor(key).node);
     }
 }
+
+test "inorderIterator" {
+    var treap = TestTreap{};
+    var nodes: [10]TestNode = undefined;
+
+    // Build the tree.
+    var i: usize = 0;
+    while (i < 10) : (i += 1) {
+        const key = @as(u64, i);
+        var entry = treap.getEntryFor(key);
+        entry.set(&nodes[i]);
+    }
+
+    // Test the iterator.
+    var iter = treap.inorderIterator();
+    i = 0;
+    while (iter.next()) |node| {
+        const key = @as(u64, i);
+        try testing.expectEqual(key, node.key);
+        i += 1;
+    }
+}
+
+test "getMin, getMax" {
+    var treap = TestTreap{};
+    var nodes: [3]TestNode = undefined;
+
+    try testing.expectEqual(treap.getMin(), null);
+    try testing.expectEqual(treap.getMax(), null);
+    { // nodes[1]
+        var entry = treap.getEntryFor(1);
+        entry.set(&nodes[1]);
+        try testing.expectEqual(treap.getMin(), &nodes[1]);
+        try testing.expectEqual(treap.getMax(), &nodes[1]);
+    }
+    { // nodes[0]
+        var entry = treap.getEntryFor(0);
+        entry.set(&nodes[0]);
+        try testing.expectEqual(treap.getMin(), &nodes[0]);
+        try testing.expectEqual(treap.getMax(), &nodes[1]);
+    }
+    { // nodes[2]
+        var entry = treap.getEntryFor(2);
+        entry.set(&nodes[2]);
+        try testing.expectEqual(treap.getMin(), &nodes[0]);
+        try testing.expectEqual(treap.getMax(), &nodes[2]);
+    }
+}
+
+test "node.{prev(),next()} with sequential insertion and deletion" {
+    // Insert order: 50, 0, 1, 2, ..., 49, 51, 52, ..., 99.
+    // Delete order: 0, 1, 2, ..., 49, 51, 52, ..., 99.
+    // Check 50's neighbors.
+    var treap = TestTreap{};
+    var nodes: [100]TestNode = undefined;
+    {
+        var entry = treap.getEntryFor(50);
+        entry.set(&nodes[50]);
+        try testing.expectEqual(nodes[50].key, 50);
```

Comment:
These arguments should be switched, since `expectEqual` takes `expected, actual`. There are a few other places where the parameters should be reversed as well (line 527, lines 591-598 are the ones I noticed).

Reply:
Done